distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -1,136 +1,137 @@
 import project.Any.Any
 import project.Data.Array.Array
+import project.Data.Array_Proxy.Array_Proxy
 import project.Data.Json.Json
+import project.Data.Json.JS_Object
 import project.Data.Locale.Locale
 import project.Data.Map.Map
+import project.Data.Numbers.Decimal
+import project.Data.Numbers.Integer
 import project.Data.Numbers.Number
 import project.Data.Range.Extensions
+import project.Data.Text.Extensions
 import project.Data.Text.Text
+import project.Data.Text.Text_Sub_Range.Text_Sub_Range
 import project.Data.Vector.Vector
 import project.Error.Error
 import project.Meta
 import project.Nothing.Nothing
 
 from project.Data.Boolean import Boolean, True, False
 
-## ALIAS To JSON
+## Converts the given value to a JSON serialized value.
+Any.to_json : Text
+Any.to_json self = Json.stringify self
 
-   Generically converts an atom into a JSON object.
-
-   The input atom is converted into a JSON object, with a `"type"` field set to
-   the atom's type name and all other fields serialized with their name as
-   object key and the value as the object value.
-
-   > Example
-     Convert a vector to JSON.
-         [1, 2, 3, 4].to_json
-Any.to_json : Json
-Any.to_json self =
+## Converts the given value to a JSON serialized value.
+Error.to_json : Text
+Error.to_json self =
+    caught = self.catch
+    error_type = ["type", "Error"]
+    error_content = ["content", caught]
+    error_message = ["message", caught.to_display_text]
+    JS_Object.from_pairs [error_type, error_content, error_message] . to_text
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Text.to_js_object : JS_Object | Text
+Text.to_js_object self = case self of
+    Text -> JS_Object.from_pairs [["type", "Text"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Number.to_js_object : JS_Object | Number
+Number.to_js_object self = case self of
+    Number -> JS_Object.from_pairs [["type", "Number"]]
+    Integer -> JS_Object.from_pairs [["type", "Integer"]]
+    Decimal -> JS_Object.from_pairs [["type", "Decimal"]]
```

Comment:
Good catch that these need to be accounted for. Does it work though? I'm not sure how it plays with `self` tbh.

Regardless if it works (well if not - then even more so) - this really feels weird to look at - ofc not something to do for now, but I think we should figure out how to distinguish defining such methods for the instance and their static variants. My intuition would see them as rather separate definitions, instead of having to do such matches.

@hubertp @kustosz as you've been close to the statics - any thoughts? Is it even possible to have a static and non-static variant of a method with same name? I'm worried it may conflict with the `Type.instance_method instance_ref args` call pattern, so I'm not sure if it is even possible. Still seems that from the 'user' (in this context: library developer) perspective, we need to be able to define _some_ methods both on instances and on the type itself.